packages/react/src/components/SelectItemGroup/SelectItemGroup-test.js

```diff
@@ -7,37 +7,56 @@
 
 import React from 'react';
 import SelectItemGroup from '../SelectItemGroup';
-import { shallow } from 'enzyme';
+import { render, screen } from '@testing-library/react';
 
 const prefix = 'cds';
 
 describe('SelectItemGroup', () => {
   describe('Renders as expected', () => {
-    const wrapper = shallow(
-      <SelectItemGroup className="extra-class" label="test" />
-    );
+    const renderSelectItemGroup = (props) =>
+      render(
+        <SelectItemGroup
+          className="extra-class"
+          data-testid="select-item-group"
+          label="testLabel"
+          {...props}
+        />
+      );
```

Comment:
Our test files usually don't follow this pattern and instead call `render` directly in each `it()` block. It's more duplication, but the benefit we've found is that it's easier to maintain as the tests grow. Changing a single test due to future API changes is simpler when there's no abstraction.

We don't have the eslint rule `testing-library/no-render-in-setup` on yet, but it's worth noting this _will still pass_ the rule, so it's not a hard blocker.

Reply:
Sure, I will refactor away all of the render* methods I have added to tests. As for the other stuff, I noticed that container was used heavily, so I started down that path. However, I felt guilty for taking a shortcut. I have been following their priority queue, and they recommend using test ids before containers.

> In the spirit of [the guiding principles](https://testing-library.com/docs/guiding-principles), it is recommended to use this only after the other queries don't work for your use case. **Using data-testid attributes do not resemble how your software is used and should be avoided if possible. That said, they are way better than querying based on DOM structure or styling css class names.** Learn more about data-testids from the blog post ["Making your UI tests resilient to change"](https://kentcdodds.com/blog/making-your-ui-tests-resilient-to-change)

I guess it doesn't explicitly state "don't use containers", but I think most of the container usage I've seen was to access DOM structure or class names. 

I used Abbey's script on one of the tests I wrote, but it felt like more work than if I just converted over the existing test. I can go back and scrub the tests I have done with her script.

I have a feeling most (if not all) of the container references in Carbon tests can be refactored to use test ids instead of containers.

Reply:
e.g. Abbey's script produces this:
```
    it('should support a custom `className` prop on the outermost element', () => {
      const { container } = render(<SelectItemGroup className="custom-class" />);
      expect(container.firstChild).toHaveClass('custom-class');
    });
```

Is not using container to get the firstChild, based on the DOM structure? When using getByTestId is higher in the priority queue?
`expect(screen.getByTestId('test-id')).toHaveClass('custom-class')`

Reply:
Hey so sorry for the delay here! Somehow this slipped out of my review queue.

Yeah the intent from that test within Abbey's script is that we want to ensure we're following the pattern we've agreed on, which is to only apply classnames to the outermost element of a component. It's one of the only test patterns where we specifically _want_ to test the DOM structure.

Technically a `data-testid` could be placed anywhere, whereas `container` is static. `container.firstChild` more confidently asserts that the outermost element is receiving the classname.